### DISCUSSION TITLE
Connection Error - 'winiomwin' - encryption error

### DISCUSSION THREAD

**Discussion by ChrisR009** [sassoftware/saspy#408] [Q&A] [answered]
Hi. Having difficulty connecting on 'winiomwin' configuration. Receiving the following errors which appear to be related to encryption (AES) mismatch between saspy and server . 

We believe we have the configuration file correct.  Any direction on how to resolve this would be appreciated. 

Thankyou.

# Using 
- Python 3.9.7
- saspy 3.7.5
- JRE 7 (java version "1.7.0_80")
- SAS 9.4 m4
- Windows Client
- Windows Server

# Error Messages
 We failed in getConnection
 None of the requested encryption algorithms are supported by both peers: AES

 SAS process has terminated unexpectedly. RC from wait was: 4294967290
 SAS Connection failed. No connection established. Double check your settings in sascfg_personal.py file.

 KeyError: 'No SAS process attached. SAS process has terminated unexpectedly.'
 ---------------------------------------------------------------------------
 KeyError                                  Traceback (most recent call last)
 h:\Documents\Stash\demo-saspy\sample-saspy.py in <module>
 ----> 14 sas=saspy.SASsession(cfgname='winiomwin')

 ~\AppData\Local\Programs\Python\Python39\lib\site-packages\saspy\sasbase.py in __init__(self, **kwargs)
     573         if self.sascfg.mode != 'HTTP':
     574            try:
 --> 575               self.pyenc = sas_encoding_mapping[self.sascei]
     576            except KeyError:
     577               logger.fatal("Invalid response from SAS on inital submission. printing the SASLOG as diagnostic")

 KeyError: 'No SAS process attached. SAS process has terminated unexpectedly.'

**Comment by tomweber-sas** [comment]
Hey @ChrisR009 would you mind moving this to an Issue? Problems and requests for help should be opened as an Issue, as opposed to being here in this discussions channel. That way when people search for similar problem/answers they will find them. 
I think the solution should be simple. Your Workspace server is configured w/ AES encryption. So to connect, you either need the encryption jars added to the iomclient directory in your saspy install (https://sassoftware.github.io/saspy/configuration.html#attn-as-of-saspy-version-3-3-3-the-classpath-is-no-longer-required), or you just need to upgrade Java to the current Java 8 which has support for this in it (then you don't need the jars).

**Comment by tomweber-sas** [comment, marked as answer]
BTW, I though I'd documented this error previously in the troubleshooting guide, but I hadn't. So, I just added it there; see number 8 here: https://sassoftware.github.io/saspy/troubleshooting.html#iom-specific-errors
Sorry that wasn't there for you when you hit this, but it there now, at least.
Tom
